### PR TITLE
docs refactor 3/n

### PR DESCRIPTION
**Pull request by williamFalcon**
The rest of the docs refactor which involves:
- organization
- content changes
- still some to dos in follow on PRs to continue cleaning up / simplifying the docs.

cc @borda @rohitgr7 @justusschock @awaelchli @tchaton

### CODE REVIEW

File: docs/source/accelerators/accelerator_prepare.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+To train on CPU/GPU/TPU without changing your code, we need to build a few good habits :)
+
+----
+
+*****************************
+Delete .cuda() or .to() calls
+*****************************
+
+Delete any calls to .cuda() or .to(device).
```
Comment (justusschock) [inline docs/source/accelerators/accelerator_prepare.rst:21]: The same also goes for .cpu() calls in most of the cases

File: docs/source/accelerators/accelerator_prepare.rst [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    # before lightning
+    def forward(self, x):
+        z = torch.Tensor(2, 3)
+        z = z.cuda(0)
+
+
+    # with lightning
+    def forward(self, x):
+        z = torch.Tensor(2, 3)
```
Comment (justusschock) [inline docs/source/accelerators/accelerator_prepare.rst:54]: wouldn't it be better to use `torch.tensor/torch.empty` with `
dtype=x.dtype
` here? this should be one operation less.

Also should we maybe make use of `self.dtype` and `self.device` here?

File: docs/source/accelerators/accelerator_prepare.rst [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    class LitModel(LightningModule):
+        def __init__(self):
+            ...
+            self.register_buffer("sigma", torch.eye(3))
+            # you can now access self.sigma anywhere in your module
+
+----
+
+***************
+Remove samplers
```
Comment (justusschock) [inline docs/source/accelerators/accelerator_prepare.rst:73]:
```suggestion
Remove distributed samplers
```
every other custom sampler still needs to stay for same behavior :)

**Review by justusschock — COMMENTED** [review]
First part of the review, not yet complete.

File: docs/source/levels/core_skills.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+.. displayitem::
+   :header: Level 5: Understand and visualize your model
+   :description: Remove bottlenecks and visualize your model
+   :button_link: ../levels/basic_level_5.html
+   :col_css: col-md-6
+   :height: 150
+   :tag: basic
+
+.. displayitem::
+   :description: Use your model for predictions.
```
Comment (rasbt) [inline docs/source/levels/core_skills.rst:60]: I would flip the 2 rows

    :description: Use your model for predictions.

and

    :header: Level 6: Predict with your model

here 

to make it consistent with the previous entries.

File: docs/source/levels/core_skills.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+   :header: Level 2: Add a validation and test set
+   :description: Add validation and test sets to avoid over/underfitting.
+   :button_link: /levels/basic_level_2.html
+   :col_css: col-md-6
+   :height: 150
+   :tag: basic
+
+.. displayitem::
+   :header: Level 3: Use pretrained models
+   :description: Learn how to use pretrained models with Lightning
```
Comment (rasbt) [inline docs/source/levels/core_skills.rst:37]: Add a period after "Lightning" for consistency with other entries.

File: docs/source/levels/core_skills.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+   :header: Level 4: Enable script parameters
+   :description: Add parameters to your script so you can run from the commandline.
+   :button_link: ../common/hyperparameters.html
+   :col_css: col-md-6
+   :height: 150
+   :tag: basic
+
+.. displayitem::
+   :header: Level 5: Understand and visualize your model
+   :description: Remove bottlenecks and visualize your model
```
Comment (rasbt) [inline docs/source/levels/core_skills.rst:53]: Add a period after "model" for consistency with other entries.

File: docs/source/levels/expert_level_24.rst [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+.. raw:: html
+
+    <div class="display-card-container">
+        <div class="row">
+
+.. Add callout items below this line
+
+.. displayitem::
+   :header: Integrate your own cluster
+   :description: Learn how to integrate your own cluster
```
Comment (rasbt) [inline docs/source/levels/expert_level_24.rst:18]: Add a period after "cluster" for consistency with other entries.

File: docs/source/levels/expert_level_24.rst [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+   :col_css: col-md-6
+   :button_link: ../clouds/cluster_expert.html
+   :height: 150
+   :tag: expert
+
+.. displayitem::
+   :header: Run on your own cloud
+   :description: Learn how to run on your Company or University private clouds.
```
Comment (rasbt) [inline docs/source/levels/expert_level_24.rst:26]: Company and University are intentionally capitalized?

File: docs/source/levels/intermediate_level_12.rst [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+   :header: Deploy with torchscript
+   :description: Optimize models for enterprise-scale production environments with torchscript.
+   :col_css: col-md-4
+   :button_link: ../deploy/production_advanced_2.html
+   :height: 180
+   :tag: advanced
+
+.. displayitem::
+   :header: Compress models for fast inference
+   :description: Compress model size for deployment with Quantization and Pruning.
```
Comment (rasbt) [inline docs/source/levels/intermediate_level_12.rst:34]: Should quantization and pruning be capitalized? Maybe we should use italics for emphasis?

File: docs/source/levels/intermediate_level_14.rst [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+   :header: Run on an on-prem cluster
+   :description: Learn to train models on a general compute cluster.
+   :col_css: col-md-4
+   :button_link: ../clouds/cluster_intermediate_1.html
+   :height: 150
+   :tag: intermediate
+
+.. displayitem::
+   :header: Run on a SLURM cluster
+   :description: Run models on a SLURM-managed cluster
```
Comment (rasbt) [inline docs/source/levels/intermediate_level_14.rst:26]: Add a period after "cluster"

File: docs/source/levels/intermediate_level_14.rst [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+   :col_css: col-md-4
+   :button_link: ../clouds/cluster_advanced.html
+   :height: 150
+   :tag: intermediate
+
+.. displayitem::
+   :header: Run with Torch Distributed
+   :description: Run models on a cluster with torch distributed.
```
Comment (rasbt) [inline docs/source/levels/intermediate_level_14.rst:34]: Maybe "torch distributed" could be shown as "`torch.distributed`" so people know it's a package and not a description.

File: docs/source/levels/intermediate_level_7.rst [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+----
+
+.. raw:: html
+
+    <div class="display-card-container">
+        <div class="row">
+
+.. displayitem::
+   :header: Prepare your code (Optional)
```
Comment (rasbt) [inline docs/source/levels/intermediate_level_7.rst:15]: Should "Optional" be capitalized?

**Comment by williamFalcon** [comment]
@ananthsub this PR merges structural changes and _some_ clean up.
Maybe let's leave the comments there so we can address in future PRs? we need to spread the doc pages to review across the team, so maybe you can take the GPU section?
cc @Borda

**Review by kaushikb11 — APPROVED** [review]

**Pull request merged**